src/dialog-renderer.js

```diff
@@ -47,7 +47,7 @@ export class DialogRenderer {
   _escapeKeyEventHandler = (e) => {
     if (e.keyCode === 27) {
       let top = this._dialogControllers[this._dialogControllers.length - 1];
-      if (top && top.settings.lock !== true) {
+      if (top && (top.settings.lock !== true && !top.settings.enableEscClose)) {
```

Comment:
I think this should be `!top.settings.enableEscClose` because when I pull this down and test it in the sample app it does not close, when I set `enableEscClose: false` it does, which seems backwards.